src/item_factory.cpp

```diff
@@ -2608,31 +2608,97 @@ void Item_factory::add_special_pockets( itype &def )
     }
 }
 
-static int parse_acc_string( const std::string &to_hit )
+namespace acc_parse
+{
+// *INDENT-OFF*
+enum class grip_val : int {
+    BAD = -1,
+    NONE,
+    SOLID,
+    WEAPON
+};
+enum class length_val : int {
+    HAND = 0,
+    SHORT,
+    LONG
+};
+enum class surface_val : int {
+    POINT = -2,
+    LINE,
+    ANY,
+    EVERY
+};
+enum class balance_val : int {
+    CLUMSY = -2,
+    UNEVEN,
+    NEUTRAL,
+    GOOD
+};
+
+template<>
+std::string enum_to_string<grip_val>( grip_val val )
 {
-    int acc = -2;
-    std::string grip, length, surface, balance;
-    size_t last_split = 0;
-    size_t split_len = to_hit.find( "_", last_split );
+	switch( val ) {
+	case grip_val::BAD: return "bad";
+	case grip_val::NONE: return "none";
+	case grip_val::SOLID: return "solid";
+	case grip_val::WEAPON: return "weapon";
+	default: break;
+	}
+	debugmsg( "Invalid grip val" );
+	abort();
+}
 
-    if( split_len != std::string::npos && split_len > 0 ) {
-        grip = to_hit.substr( last_split, split_len );
-    }
-    last_split = split_len + 1;
-    split_len = to_hit.find( "_", last_split );
-    if( split_len != std::string::npos && split_len > 0 ) {
-        length = to_hit.substr( last_split, split_len - last_split );
-    }
-    last_split = split_len + 1;
-    split_len = to_hit.find( "_", last_split );
-    if( split_len != std::string::npos && split_len > 0 ) {
-        surface = to_hit.substr( last_split, split_len - last_split );
-    }
-    last_split = split_len + 1;
-    split_len = to_hit.size();
-    if( last_split < split_len ) {
-        balance = to_hit.substr( last_split, split_len );
-    }
+template<>
+std::string enum_to_string<length_val>( length_val val )
+{
+        switch( val ) {
+        case length_val::HAND: return "hand";
+        case length_val::SHORT: return "short";
+        case length_val::LONG: return "long";
+        default: break;
+        }
+        debugmsg( "Invalid length val" );
+        abort();
+}
+
+template<>
+std::string enum_to_string<surface_val>( surface_val val )
+{
+        switch( val ) {
+        case surface_val::POINT: return "point";
+        case surface_val::LINE: return "line";
+        case surface_val::ANY: return "any";
+        case surface_val::EVERY: return "every";
+        default: break;
+        }
+        debugmsg( "Invalid surface val" );
+        abort();
+}
+
+template<>
+std::string enum_to_string<balance_val>( balance_val val )
+{
+        switch( val ) {
+        case balance_val::CLUMSY: return "clumsy";
+        case balance_val::UNEVEN: return "uneven";
+        case balance_val::NEUTRAL: return "neutral";
+        case balance_val::GOOD: return "good";
+        default: break;
+        }
+        debugmsg( "Invalid balance val" );
+        abort();
+}
```

Comment:
this needs to be in the `io` namespace. usually i do `// *INDENT-OFF*` as well.

Reply:
Thanks, fixed.